src/Grpc.AspNetCore.Server/Internal/CallHandlers/DuplexStreamingServerCallHandler.cs

```diff
@@ -52,11 +68,45 @@ protected override async Task HandleCallAsyncCore(HttpContext httpContext)
 
                 service = activator.Create();
 
-                await _invoker(
-                    service,
-                    new HttpContextStreamReader<TRequest>(serverCallContext, Method.RequestMarshaller.Deserializer),
-                    new HttpContextStreamWriter<TResponse>(serverCallContext, Method.ResponseMarshaller.Serializer),
-                    serverCallContext);
+                if (_interceptors == null)
+                {
+                    await _invoker(
+                        service,
+                        new HttpContextStreamReader<TRequest>(serverCallContext, Method.RequestMarshaller.Deserializer),
+                        new HttpContextStreamWriter<TResponse>(serverCallContext, Method.ResponseMarshaller.Serializer),
+                        serverCallContext);
+                }
+                else
+                {
+                    DuplexStreamingServerMethod<TRequest, TResponse> resolvedInvoker = (requestStream, responseStream, resolvedContext) =>
+                    {
+                        return _invoker(
+                        service,
+                        requestStream,
+                        responseStream,
+                        resolvedContext);
+                    };
+
+                    // The list is reversed during construction so the first interceptor is invoked first
+                    for (var i = _interceptors.Count - 1; i >= 0; i--)
+                    {
+                        var registration = _interceptors[i];
```

Comment:
Can 93-100 be moved to a protected method on the base class and reused in each call handler?